sklearn/preprocessing/tests/test_data.py

```diff
@@ -1217,7 +1218,7 @@ def test_normalize():
 
 
 def test_binarizer():
-    X_ = np.array([[1, 0, 5], [2, 3, -1]])
+    X_ = np.array([[1, 0, 5], [2, 3, -1]], dtype=np.float64)
```

Comment:
I suggest we change `X_bin is not X` with a comment stating that why the copy is triggered instead of changing it here.